### ISSUE TITLE
Pull searchees from qBittorrent API (BT_Backup gone)

### ISSUE THREAD

**Issue by KyleSanderson** [cross-seed/cross-seed#365]
https://github.com/qbittorrent/qBittorrent/discussions/16768

**Comment by KyleSanderson** [comment]
BT_Backup is no longer in-use, and they're using SQLite (torrents.db) with WAL to store session state.

**Comment by Slikkster** [comment]
Since you are using the qbittorrent API to inject the torrents, why not also use it for checking for existing torrents instead of BT_backup? Then it wouldn't matter how qbittorrent stores information on the disk.

**Comment by dynizzle** [comment]
I setup cross-seed on Unraid for the first time today and I think I'm running into this as well. Whenever I add torrents it doesn't even try to do a lookup but by adding the [run external program on completion](https://www.cross-seed.org/docs/basics/daemon#qbittorrent) for new torrents that are added, it works. Doesn't really help for new releases that don't exist on all my trackers yet.

What's weird is I still see new torrents being added in BT_Backup folder so I don't know if that's the issue or not. I'm thinking of just downgrading qBittorrent to <= 4.3 to see if that fixes it.

**Comment by mmgoodnow** [comment]
That's WAD then. It doesn't react to new torrents automatically

**Comment by dynizzle** [comment]
Ah OK. So with Daemon mode, is any way to force a full search of all current torrents without scripting out all the API calls manually?

**Comment by Slikkster** [comment]
Yes, just run the 'cross-seed search' command. Even if the daemon is already running, it will run the search (at least it did for me). If you're using docker, do 'docker exec -it <container> cross-seed search'

**Comment by mmgoodnow** [comment]
please take this conversation to Discussions or Discord.

**Comment by KyleSanderson** [comment]
They've added a new endpoint, `api/v2/torrents/export`, that can export the full torrent file from the client. So what could probably be done on the cross-seed side of things, is scan the list, then export the torrent files it doesn't have based on the infohash and store it in the cache.

**Comment by KyleSanderson** [comment]
Wrote this dumb thing https://github.com/KyleSanderson/sessionerr as a workaround

**Comment by onedr0p** [comment]
This would be great if cross seed was updated to support qb with sqlite. SQLite with WAL has significantly improve startup and loading times for me. I just went thru trying to set up cross-seed but to be hit by this issue, if y'all need someone to help test I'm here. 👍

**Comment by MrMarble** [comment]
Is there any update on this? It has been on ready for a while now

**Comment by KyleSanderson** [comment]
> Is there any update on this? It has been on ready for a while now. I'm wiling to do some testing if it's needed

Does Sessionerr not work for you?

**Comment by onedr0p** [comment]
I must be out of the loop, what is Sessionerr??

**Comment by MrMarble** [comment]
> I must be out of the loop, what is Sessionerr??

https://github.com/cross-seed/cross-seed/issues/365#issuecomment-1424880513
Looking at the code seems like it pulls the torrents using the qbt api and creates a BT_Backup folder to use with cross-seed

**Comment by onedr0p** [comment]
I'll wait for official support, however if @KyleSanderson properly documents their project I might give it a shot.

**Comment by mmgoodnow** [comment]
I've got a branch in progress working on this but haven't had a lot of time lately.

**Comment by KyleSanderson** [comment]
> > I must be out of the loop, what is Sessionerr??
> 
> [#365 (comment)](https://github.com/cross-seed/cross-seed/issues/365#issuecomment-1424880513) Looking at the code seems like it pulls the torrents using the qbt api and creates a BT_Backup folder to use with cross-seed
> 
> > > Is there any update on this? It has been on ready for a while now. I'm wiling to do some testing if it's needed
> > 
> > 
> > Does Sessionerr not work for you?
> 
> Tbh I did not try it, I might do, but I'll prefer official support to the new version instead of having something running in the middle

Well sure. It's kind of meant for adults 😆. If you're comfortable waiting another year you're free to do that, if you wanted a solution 6 months ago, it's been sitting there freely available for usage to get the aforementioned party out of their jam.

**Comment by onedr0p** [comment]
@KyleSanderson I don't understand your attitude it is very off putting. If you want _adults_ to use this would it be that hard to add a README with instructions?

**Comment by 0dragosh** [comment]
@KyleSanderson thanks for letting us know that you consider an undocumented, unpackaged mess of 3 Go files a project meant for adults, will stay clear of it!

**Comment by KyleSanderson** [comment]
> @KyleSanderson I don't understand your attitude... it is very off putting. If you want _adults_ to use your projects would it be that hard to add a README with instructions and share the knowledge that only you possess?

So, to clarify for the group of you. You've found a reported issue on the repository, with a linked fix, but it's not documented and there's a refusal to open the single 80sloc file with switches clearly at the top. Instead of asking for help, you've decided to blame the author for not having a public readme, and claim you're a victim. It's a fix exclusively for this problem.

The audacity here is absolutely outrageous, and I'm glad it's being used by some, and not by others. Good luck and have fun.

**Comment by onedr0p** [comment]
@KyleSanderson I clearly asked for a README with how to set it up, sorry if that wasn't clear. I don't think asking here how to configure Sessionerr is the right place.

@mmgoodnow can you please re-open this issue. It is clearly a feature request for cross-seed to support, not an undocumented tool created by @KyleSanderson
